spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepositoryIntegrationTests.java

```diff
@@ -503,6 +504,42 @@ public void shouldSetTransportConfigCallback() throws IOException {
 		assertNotNull(repository.getTransportConfigCallback());
 	}
 
+	@Test
+	public void testShouldReturnEnvironmentFromLocalBranchInCaseRemoteDeleted() throws Exception {
+		JGitConfigServerTestData testData = JGitConfigServerTestData
+				.prepareClonedGitRepository(TestConfiguration.class);
+
+		String branchToDelete = "branchToDelete";
+		testData.getServerGit().getGit().branchCreate().setName(branchToDelete).call();
+
+		Environment environment = testData.getRepository().findOne("bar", "staging", "branchToDelete");
+		assertNotNull(environment);
+
+		testData.getServerGit().getGit().branchDelete().setBranchNames(branchToDelete).call();
+		testData.getRepository().findOne("bar", "staging", "branchToDelete");
```

Comment:
Shouldn't the result be set to `environment`?

Reply:
Nice catch! Sorry for that - fixed now.
In any case main point is not to get ```NoSuchLabelException```